resources/files/sacct/nginx.conf

```diff
@@ -0,0 +1,79 @@
+user nginx;
+worker_processes auto;
+error_log /var/log/nginx/error.log;
+pid /run/nginx.pid;
+
+# Load dynamic modules. See /usr/share/doc/nginx/README.dynamic.
+include /usr/share/nginx/modules/*.conf;
+
+events {
+    worker_connections 1024;
+}
+
+http {
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /var/log/nginx/access.log  main;
+
+    sendfile            on;
+    tcp_nopush          on;
+    tcp_nodelay         on;
+    keepalive_timeout   65;
+    types_hash_max_size 4096;
+
+    include             /etc/nginx/mime.types;
+    default_type        application/octet-stream;
+
+    # Load modular configuration files from the /etc/nginx/conf.d directory.
+    # See http://nginx.org/en/docs/ngx_core_module.html#include
+    # for more information.
+    include /etc/nginx/conf.d/*.conf;
+
+# Settings for a TLS enabled server.
+
+    server {
+        listen       443 ssl http2;
+        listen       [::]:443 ssl http2;
+        server_name  _;
+        root         /usr/share/nginx/html;
+
+        location / {
+            proxy_pass                  http://slurmrestd;
+
+#           auth_request                /validate/;
+#           auth_request_set            $user_name $upstream_http_x_slurm_user_name;
+#           auth_request_set            $user_token $upstream_http_x_slurm_user_token;
+#           proxy_set_header            X-SLURM-USER-NAME $user_name;
+#           proxy_set_header            X-SLURM-USER-TOKEN $user_token;
+            proxy_set_header Host       $host;
+            proxy_set_header X-Real-IP  $remote_addr;
+            proxy_set_header            X-Forwarded-Port $server_port;
+            proxy_pass_request_headers  on;
+        }
+
+        ssl_certificate             /etc/ssl/certs/nginx-selfsigned.crt;
+        ssl_certificate_key         /etc/ssl/certs/nginx-selfsigned.key;
+        ssl_client_certificate      /etc/ssl/certs/ca.crt;
+        ssl_session_cache           shared:SSL:1m;
+        ssl_session_timeout         10m;
+        ssl_ciphers                 HIGH:!aNULL:!MD5;
+        ssl_prefer_server_ciphers   on;
+
+        # Load configuration files for the default server block.
+        include /etc/nginx/default.d/*.conf;
+
+        error_page 404 /404.html;
+            location = /40x.html {
+        }
+
+        error_page 500 502 503 504 /50x.html;
+            location = /50x.html {
+        }
+    }
+
+    upstream slurmrestd {
+        server 0.0.0.0:8080;
```

Comment:
**question**: I am not an nginx champion.. is it here where the proxy happens?

Reply:
This is what directs the HTTPS proxy to the Slurm REST API endpoint, which only supports HTTP. The HTTPS server block is the actual proxy (Charlie helped me out here so he may need to correct me).

Reply:
For added security and performance you could run the slurmrestd service on a unix socket rather than on the http port. This will avoid the risk of binding to existing ports, mistakenly exposing an open http port and can be easily configured to allow access only to the user that runs the nginx server

Reply:
Will bring this up with AppSec - Charlie said there might be a case to allow users to access the API outside of PCM